packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -239,9 +245,23 @@ export function cancelPrompt() {
   store.setState({ ...state, prompt: null });
 }
 
-export function closeWorkspace(name: string, ignoreChanges: boolean) {
+export function closeWorkspace(name: string, ignoreChanges: boolean, onCloseWorkspace?: () => void) {
   const store = getWorkspaceStore();
   const promptCheckFcn = getPromptBeforeClosingFcn(name);
+  const updateStoreWithClosedWorkspace = () => {
+    const state = store.getState();
+    const newOpenWorkspaces = state.openWorkspaces.filter((w) => w.name !== name);
+
+    store.setState({
+      ...state,
+      prompt: null,
+      openWorkspaces: newOpenWorkspaces,
+    });
+
+    if (onCloseWorkspace && typeof onCloseWorkspace === 'function') {
+      onCloseWorkspace?.();
+    }
```

Comment:
I think this call should come before the state manipulation stuff.

Reply:
I'm sorry, but which state management are we talking about? 

The workspace's state management for closing all the workspaces is being done from here only.

I'm not clear about your point.

Reply:
Move lines 261-263 so they are the first thing done in this function. Probably wrap them in a `try / finally` too.